crates/driver/src/infra/blockchain/contracts.rs

```diff
@@ -20,14 +20,14 @@ pub struct Contracts {
 
     /// The domain separator for settlement contract used for signing orders.
     settlement_domain_separator: eth::DomainSeparator,
-    cow_amm_registry: cow_amm::Registry,
 
     /// Single router that supports multiple flashloans in the
     /// same settlement.
     // TODO: make this non-optional when contracts are deployed
     // everywhere
     flashloan_router: Option<FlashLoanRouter::Instance>,
     balance_helper: contracts::support::Balances,
+    cow_amm_helper_by_factory: HashMap<eth::ContractAddress, eth::ContractAddress>,
```

Comment:
nit: small comment explicitly stating which address is the key and which the value would be nice since we just the same type for key and value here.